python/tvm/driver/tvmc/transform.py

```diff
@@ -21,7 +22,88 @@
 from tvm.driver.tvmc import TVMCException
 
 
-def convert_graph_layout(mod, desired_layout):
+def generate_mixed_precision_rule(acc_dtype):
+    def _mixed_precision_rule(call_node: "relay.Call", mixed_precision_type: str):
+        return [
+            relay.transform.mixed_precision.MIXED_PRECISION_ALWAYS,
+            acc_dtype,
+            mixed_precision_type,
+        ]
+
+    return _mixed_precision_rule
+
+
+class MixedPrecision(object):
+    """Temporarily changes attr of ops to enable required precision."""
+
+    def __init__(self, ops, acc_type):
+        """Saves the required info for RAII pattern usage.
+
+        Parameters
+        ----------
+        ops : list
+            list of operators
+        acc_type: str
+            Output or accumulation precision to be used.
+        """
+        self.older_attr = {}
+        self.ops = ops
+        self.acc_type = acc_type
+        self.attr_key = "FTVMMixedPrecisionConversionType"
+
+    def __enter__(self):
+        for op_name in self.ops:
+            op = relay.op.get(op_name)
+            self.older_attr[op_name] = op.get_attr(self.attr_key)
+            op.reset_attr(self.attr_key)
+            op.set_attr(self.attr_key, generate_mixed_precision_rule(self.acc_type))
+        return self
+
+    def __exit__(self, ptype, value, trace):
+        for op_name in self.ops:
+            op = relay.op.get(op_name)
+            op.reset_attr(self.attr_key)
+            if self.older_attr[op_name]:
+                op.set_attr(self.attr_key, self.older_attr[op_name])
+
+
+def convert_to_mixed_precision(mod, ops=None, calculation_type="float16", acc_type="float16"):
+    """Converts the operator datatypes
+
+    Parameters
+    ----------
+    mod : tvm.IRModule
+        The relay module to convert.
+    ops : list
+        List of operators to be precision converted.
+    calculation_type: str
```

Comment:
this parameter is not used. need to remove from parameters of `convert_to_mixed_precision` or pass it to `ToMixedPrecision`